atomic_reactor/plugins/pre_pin_operator_digest.py

```diff
@@ -229,10 +249,70 @@ def replace_registry(self, image):
             return image
         return self._replace(image, registry=self.registry_replace[image.registry])
 
+    # TODO: rename, because it can also replace namespace
+    def replace_repo(self, image):
+        """
+        Replace image repo based on OSBS site/user configuration for image registry
+
+        :param image: ImageName
+        :return: ImageName
+        """
+        package_mapping = self._get_package_mapping(image.registry)
+        if package_mapping is None:  # repo_replacements not configured for registry
+            return image
+
+        replacement = package_mapping.get(image.repo)
+
+        if replacement is None:
+            raise RuntimeError(
+                "repo_replacements configured for registry but repo is missing: {}".format(image)
+            )
+        elif isinstance(replacement, list):
+            replacements = ", ".join(replacement)
+            raise RuntimeError(
+                "Multiple options found when replacing repo in {}: {}".format(image, replacements)
+            )
+
+        # TODO: replacement can (probably should) be namespace/repo
+        # what should happen if replacement is only repo? drop original namespace? or keep?
+        # note: original pullspec will always have namespace here, because:
+        #   this replacement only happens if configured for registry
+        #   ImageName parses foo/bar as namespace/repo
+        #   => to have registry, you must also have namespace
+        replacement = ImageName.parse(replacement)
+        return self._replace(image, namespace=replacement.namespace, repo=replacement.repo)
+
     def _replace(self, image, registry=_KEEP, namespace=_KEEP, repo=_KEEP, tag=_KEEP):
         return ImageName(
             registry=image.registry if registry is _KEEP else registry,
             namespace=image.namespace if namespace is _KEEP else namespace,
             repo=image.repo if repo is _KEEP else repo,
             tag=image.tag if tag is _KEEP else tag,
         )
+
+    def _get_package_mapping(self, registry):
+        if registry in self.package_mapping_by_registry:
+            return self.package_mapping_by_registry[registry]
+
+        mapping_file = self.package_mapping_files.get(registry)
+        if mapping_file is not None:
+            mapping = read_yaml_from_file_path(mapping_file, "schemas/package_mapping.json")
+        else:
+            mapping = None
+
+        user_mapping = self.user_package_mappings.get(registry)
+
+        if mapping is not None or user_mapping is not None:
+            mapping = mapping or {}
+            user_mapping = user_mapping or {}
+
+            # Mapping file is [package => list of repos], user mapping is [package => repo],
+            # convert to [package => repo] where possible
+            for repo, replacements in mapping.items():
+                if len(replacements) == 1:
+                    mapping[repo] = replacements[0]
```

Comment:
I'm not a fan of mixing various types in dict

Reply:
ok, will use list everywhere